src/polymatheia/data/reader.py

```diff
@@ -322,3 +355,56 @@ def __iter__(self):
         if self._file.seekable():
             self._file.seek(0)
         return NavigableDictIterator(iter(DictReader(self._file)))
+
+
+class SRUExplainRecordReader(object):
+    """The class:`~polymatheia.data.reader.SRUExplainRecordReader` is a container for SRU Explain Records."""
+
+    def __init__(self, url):
+        """Construct a new class:`~polymatheia.data.reader.SRUExplainRecordReader`.
+
+        :param url: The base URL of the SRU server
+        :type url: ``str``
+        """
+        self._url = url
+        self._explain = SRUpy(self._url).explain()
+        self.schemas = [(schema["@name"], schema.title)
+                        for schema in NavigableDict(self._explain).explain.schemaInfo.schema]
+        self.echo = NavigableDict(self._explain.echo)
+
+    def __iter__(self):
+        """Return this :class:`~polymatheia.data.reader.SRUExplainRecordReader`` as the iterator."""
+        return NavigableDictIterator(iter(self._explain),
+                                     mapper=lambda record: {record[0]: record[1]}
+                                     )
+
+
+class SRURecordReader(object):
+    """The :class:`~polymatheia.data.reader.SRURecordReader` is an iteration container for Records fetched via SRU.
+
+    The underlying library (SRUpy) automatically handles the continuation parameters, allowing for simple iteration.
+    """
+
+    def __init__(self, url, **kwargs):
+        """Construct a new :class:`~polymatheia.data.reader.SRURecordReader`.
+
+        :param url: The base URL of the SRU endpoint
+        :type url: ``str``
+        :type plain: ``boolean``
+        :param kwargs: Requests parameters that will be sent to the SRU server
+        """
+        self._url = url
+        self._kwargs = kwargs
+        self._records = SRUpy(self._url).get_records(**self._kwargs)
```

Comment:
This causes a duplicate query to be sent. Obviously if the user wants to know how many records before iterating, this is necessary. "result_count" (see below) should probably be a getter, so that the duplicate query is only sent if the user actually tries to read the value of result_count. See #11 for the same issue on the EuropeanaSearchReader.

Reply:
Not sure if I understood you correctly, please check the new commits, thank you!

Reply:
Yes, that's fine. I will refactor this at a later point, when I do #11.